src/pages/api/multipleProductValidity.ts

```diff
@@ -0,0 +1,130 @@
+        };
+        response.push(product);
+    }
+    return response;
+};
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        if (!isSessionValid(req, res)) {
+            throw new Error('Session is invalid.');
+        }
+        const cookies = new Cookies(req, res);
+        const operatorCookie = unescape(decodeURI(cookies.get(OPERATOR_COOKIE) || ''));
+        const fareZoneCookie = unescape(decodeURI(cookies.get(CSV_ZONE_UPLOAD_COOKIE) || ''));
+        const singleOperatorCookie = unescape(decodeURI(cookies.get(PERIOD_SINGLE_OPERATOR_SERVICES_COOKIE) || ''));
+        const periodTypeCookie = unescape(decodeURI(cookies.get(PERIOD_TYPE_COOKIE) || ''));
+        const numberOfProductsCookie = unescape(decodeURI(cookies.get(NUMBER_OF_PRODUCTS_COOKIE) || ''));
+        const multipleProductCookie = unescape(decodeURI(cookies.get(MULTIPLE_PRODUCT_COOKIE) || ''));
+
+        if (
+            numberOfProductsCookie === '' ||
+            multipleProductCookie === '' ||
+            periodTypeCookie === '' ||
+            (operatorCookie === '' && (fareZoneCookie === '' || singleOperatorCookie === ''))
+        ) {
+            throw new Error('Necessary cookies not found for multiple product validity page');
+        }
+
+        const numberOfProducts = Number(JSON.parse(numberOfProductsCookie).numberOfProductsInput);
+        const productsFromCookie: Product[] = JSON.parse(multipleProductCookie);
+        const products = addErrorsIfInvalid(req, numberOfProducts, productsFromCookie);
+        const newMultipleProductCookieValue = JSON.stringify(products);
+        setCookieOnResponseObject(getDomain(req), MULTIPLE_PRODUCT_COOKIE, newMultipleProductCookieValue, req, res);
+
+        if (products.some(el => el.productValidity?.error !== '')) {
+            redirectTo(res, '/multipleProductValidity');
+            return;
+        }
+
+        let props = {};
+        const { operator, uuid, nocCode } = JSON.parse(operatorCookie);
+        const { periodTypeName } = JSON.parse(periodTypeCookie);
+
+        if (fareZoneCookie) {
+            const { fareZoneName } = JSON.parse(fareZoneCookie);
+            const atcoCodes: string[] = await getCsvZoneUploadData(uuid);
+            const zoneStops: Stop[] = await batchGetStopsByAtcoCode(atcoCodes);
+
+            if (zoneStops.length === 0) {
+                throw new Error(`No stops found for atcoCodes: ${atcoCodes}`);
+            }
+
+            props = {
+                zoneName: fareZoneName,
+                stops: zoneStops,
+            };
+        }
+
+        if (singleOperatorCookie) {
+            const { selectedServices } = JSON.parse(singleOperatorCookie);
+            props = {
+                selectedServices,
+            };
+        }
+
+        const multipleProductPeriod: DecisionData = {
+            operatorName: operator,
+            type: periodTypeName,
+            nocCode,
+            products,
```

Comment:
This will be passing through unnecessary data to the matching json like the ids and productValidity will include the error. We should only need: `productName`, `productPrice`, `productDuration` and `productValidity`

Reply:
We'll also need to look into updating the `periodValidity` api so that it passes through the single product in the same format, ie. an array of one product with these values